### PR TITLE
Update CompatibilitySuppressions after updating APICompat that uses references

### PR DESCRIPTION
**Pull request by joperezr**
Updating the Compatibility package used in dotnet/runtime repo and updating the CompatibilitySuppresion.xml files now that ApiCompat is able to follow type-forwards. This now allow us to remove some suppressions that were just due to not finding the types to the forwarded assembly, as well as add new suppressions for breaking changes to types that are forwarded.

### CODE REVIEW

File: src/libraries/System.ServiceProcess.ServiceController/src/CompatibilitySuppressions.xml
Comment (joperezr) [reply src/libraries/System.ServiceProcess.ServiceController/src/CompatibilitySuppressions.xml:9]: > Based on the diff and the amount of suppressions being added in comparison to what's being deleted, what do we gain by merging this in now instead of waiting until your mentioned change is merged and consumable?

By enabling this, we first get coverage of a current blind spot (testing facades) and we also get us closer to our goal of getting all APICompat warnings with a very small cost (which is just temporary) which is these CP1002 errors which may or may not be relevant. We already saw in other places (like your AccessControl PR) that PackageValidation has helped catch interesting issues on the build, so I'd rather check this in and have some progress made as opposed to waiting for the full solution and miss a) feedback on this feature since we are shipping it in 6.0, b) coverage for full facades, c) coverage of other potential APICompat issues with our build.

Comment (ViktorHofer) [reply src/libraries/System.ServiceProcess.ServiceController/src/CompatibilitySuppressions.xml:9]: That sounds like the right set of reasons to get this in 👍

**Review by ViktorHofer — APPROVED** [review]
Approving under the assumption that some of the suppression files noise is just temporary and the files will be cleaned-up with a follow-up change.

**Comment by joperezr** [comment]
Errors in CI are unrelated. Merging to unblock upcoming maestro consumption PRs

**Comment by ViktorHofer** [comment]
@joperezr please link to existing issues or file new ones for the unrelated failures in the PR, according to our [PR guide](https://github.com/dotnet/runtime/blob/main/docs/pr-guide.md).

File: src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/CompatibilitySuppressions.xml
Comment (safern) [inline src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/CompatibilitySuppressions.xml:9]: Why does this one still need these suppressions? I thought that was fixed with your change?

Comment (safern) [reply src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/CompatibilitySuppressions.xml:9]: Disregard, this was the first PR that caught the issue with these references.

Comment (joperezr) [reply src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/CompatibilitySuppressions.xml:9]: Yup, haven’t updated the repo with the new dependency’s logic yet

Comment (ViktorHofer) [reply src/libraries/Microsoft.Extensions.Hosting.WindowsServices/src/CompatibilitySuppressions.xml:9]: @joperezr can we please do that for main? Would be great to get the suppression files as clean as possible.

**Pull request merged**
